// ng2-materialize/dist/sidenav/sidenav.component.js
import { Component, Input, Renderer, } from '@angular/core';
var MzSidenavComponent = (function () {
    function MzSidenavComponent(renderer) {
        this.renderer = renderer;
    }
    MzSidenavComponent.prototype.ngAfterViewInit = function () {
        this.initCollapseButton();
        this.initCollapsibleLinks();
    };
    MzSidenavComponent.prototype.ngOnDestroy = function () {
        $("#" + this.collapseButtonId).sideNav('destroy');
    };
    MzSidenavComponent.prototype.initCollapseButton = function () {
        if (!this.collapseButtonId) {
            return;
        }
        var collapseButton = $("#" + this.collapseButtonId)[0];
        // Make collapse button visible on all resolution if side navigation is not fixed
        if (!this.fixed) {
            this.renderer.setElementClass(collapseButton, 'show-on-large', true);
        }
        // Add data-activates to collapse button
        this.renderer.setElementAttribute(collapseButton, 'data-activates', this.id);
        // Initialize collapsible button for side navigation
        $(collapseButton).sideNav({
            closeOnClick: this.closeOnClick || false,
            edge: this.edge || 'left',
            menuWidth: isNaN(this.width) ? 300 : this.width,
        });
    };
    MzSidenavComponent.prototype.initCollapsibleLinks = function () {
        // Initialize collapsible elements
        $('.collapsible').collapsible();
    };
    return MzSidenavComponent;
}());
export { MzSidenavComponent };
MzSidenavComponent.decorators = [
    { type: Component, args: [{
                selector: 'mz-sidenav',
                template: "<ul class=\"side-nav {{ backgroundClass }}\" [attr.id]=\"id\" [class.fixed]=\"fixed\"><ng-content></ng-content></ul>",
            },] },
];
/** @nocollapse */
MzSidenavComponent.ctorParameters = function () { return [
    { type: Renderer, },
]; };
MzSidenavComponent.propDecorators = {
    'backgroundClass': [{ type: Input },],
    'closeOnClick': [{ type: Input },],
    'collapseButtonId': [{ type: Input },],
    'edge': [{ type: Input },],
    'fixed': [{ type: Input },],
    'id': [{ type: Input },],
    'width': [{ type: Input },],
};
